File: TaskScope.BLL/Models/Dtos/Tasks/UpdateTaskDto.cs
using TaskScope.DAL.Enums;

namespace TaskScope.BLL.Models.Dtos.Tasks;

public class UpdateTaskDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public TaskPriority Priority { get; set; }
    public TaskEntityStatus Status { get; set; }
    public DateTime? DueDate { get; set; }
    public DateTime? CompletedAt { get; set; }
}

File: taskscope-client/src/models/UpdateTaskDto.ts
import { TaskEntityStatus } from "./enums/TaskEntityStatus";
import { TaskPriority } from "./enums/TaskPriority";

export interface UpdateTaskDto {
    id: string;
    title: string;
    description?: string;
    priority: TaskPriority;
    status: TaskEntityStatus;
    dueDate?: string;
    completedAt?: string;
}
